File: JAVA/237 - Delete Node in a Linked List.java
/*
 237. Delete Node in a Linked List (E)

 Approach 1 : 
 Time: O(1)
 Space: O(1)

*/

/* Approach 1 */
class Solution {
    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        node.next = node.next.next;
    }
}

File: C++/237-Delete_Node_in_a_Linked_List.cpp
/*
 237. Delete Node in a Linked List (E)

 Notes:
    Replace the given node value with the next node value, link its next to next next node.
    Don't forget to delete the node using 'delete'
*/

class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode* tmp = node->next;
        node->val = node->next->val;
        node->next = node->next->next;        
        delete tmp;
    }
};